### PR TITLE
Fixes #1304 - Upgrades python modules for the Project

### LINKED ISSUE #1304: Upgrade Flask to 0.12
We're currently on 0.10.1

http://flask.pocoo.org/docs/0.12/upgrading/

### PR DESCRIPTION
**Pull request by karlcow**
- Move to Flask 0.12
- Upgrade most of the modules which needed it
- Fixes two unnecessary import 
- Fixes code with regard to the modules upgrade.

### CODE REVIEW

**Comment by karlcow** [comment]
As said in the comments, Functionnal tests needs to be run.
r? @miketaylr

**Comment by karlcow** [comment]
hmmm fails on travis, but not on my local run.

On travis.
```
.F.F...............................E...

ERROR: Test that we are receiving the appropriate text file.
FAIL: API issue for a non existent number returns JSON 404.
FAIL: API access to labels without auth returns JSON 200.
```

On my local instance:

```
→ git status
On branch 1304/1
nothing to commit, working directory clean
(webcompatcom) 09:17:39 ~/code/webcompat.com
→ nosetests
/Users/karl/.virtualenvs/webcompatcom/lib/python2.7/site-packages/flask_limiter/extension.py:124: UserWarning: Use of the default `get_ipaddr` function is discouraged. Please refer to https://flask-limiter.readthedocs.org/#rate-limit-domain for the recommended configuration
  UserWarning
.......................................
----------------------------------------------------------------------
Ran 39 tests in 10.216s

OK
```

**Comment by karlcow** [comment]
Edited the test title for rate limit because `ERROR: Test that we are receiving the appropriate text file.` is not very explicit 😃

File: tests/test_urls.py [hunk 1]
```diff
@@ -99,7 +99,7 @@ def test_tools_cssfixme(self):
         self.assertEqual(rv.status_code, 200)
 
     def test_rate_limit(self):
-        '''Test that we are receiving the appropriate text file.'''
+        '''Rate Limit URI sends 200 OK and starts with Current user.'''
```
Comment (miketaylr) [inline tests/test_urls.py]: Thanks!

**Comment by miketaylr** [comment]
LGTM, let's see how Travis handles it. We'll def. deploy this to staging and do some extra testing there as well (only concern is Pillow dependencies).

**Comment by miketaylr** [comment]
https://travis-ci.org/webcompat/webcompat.com/builds/196676452 merge build is green. Thanks again @karlcow! 💟

**Pull request merged**